src/actions/loginUsers.ts

```diff
@@ -35,12 +52,25 @@ export const signUp = async (formData: FormData) => {
     password: formData.get('password') as string
   }
 
-  const { error } = await supabase.auth.signUp(data)
+  const { data: authData, error: authError } = await supabase.auth.signUp(data)
 
-  if (error) {
+  if (authError) {
     redirect('/error')
   }
 
+  const { error: profileError } = await supabase
+        .from('profiles')
+        .insert([
+          {
+            user_id: authData!!.user!!.id, // Extract the user ID from auth table update
+            username: formData.get('username') ? (formData.get('username') as string) : undefined //Think this makes it optional idk lel
```

Comment:
dont insert undefined at all.

for now assume user will 100% enter this. We will force them too in bit.

Reply:
actually you already have required on the input so its fine. Just tell typescript you have it, it doesn't know since the `get` method is runtime.

we will add zod package in bit though to do this type safely.